java/Ice/greeter/server/src/main/java/com/zeroc/demos/greeter/Server.java

```diff
@@ -24,9 +25,22 @@ public static void main(String[] args) {
             adapter.activate();
             System.out.println("Listening on port 4061...");
 
-            // Wait until the user presses Ctrl+C.
+            // Wait until the communicator is shut down. Here, this occurs when the user presses Ctrl+C.
             communicator.waitForShutdown();
-            System.out.println("Caught Ctrl+C, exiting...");
         }
     }
+
+    private static void shutdownCommunicatorOnCtrlC(Communicator communicator, Thread cleanupThread) {
```

Comment:
I don't like cleanupThread name very much, what about using just mainThread?

Reply:
I also prefer mainThread. It's not obvious from the name cleanupThread that your intention is that a user would do cleanup in the main thread, since we don't do any. Especially since the contention in Java seems to be that you would do the cleanup in this hook.

Reply:
Like Jose suggested, I would just call it mainThread.